src/mpl/src/gpu/mpl_gpu_cuda.c

```diff
@@ -4,10 +4,22 @@
  */
 
 #include "mpl.h"
+#include <assert.h>
 
 #define CUDA_ERR_CHECK(ret) if (unlikely((ret) != cudaSuccess)) goto fn_fail
 #define CU_ERR_CHECK(ret) if (unlikely((ret) != CUDA_SUCCESS)) goto fn_fail
 
+typedef struct {
+    uintptr_t remote_base_addr;
+    uintptr_t dev_addr;
+    uintptr_t offset;
+} gpu_ipc_handle_obj_s;
+
+static MPL_gavl_tree_t *cuda_ipc_handle_trees;
+static int node_local_size;
+static int self_local_rank;
```

Comment:
Can you rename this to `local_rank`?  `self_local_rank` makes it sound like the rank in `MPI_COMM_SELF`, which is always `0`.